From 8fb97a6f6d86d6eedccd69ce50078e441f7a0276 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Wed, 2 Nov 2011 13:00:58 +0100
Subject: [PATCH 3/3] intel-hda: fix stream search

commit ba43d28916c4f51c19bd7366089155ce81bee058 introduces a bug:
The stream-not-found case doesn't error out any more, instead the
code silently uses the first stream.  Fix it.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>

bugzilla: 740493 - audio playing doesn't work when sound recorder is
          opened
upstream: http://patchwork.ozlabs.org/patch/123258/
Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/intel-hda.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/intel-hda.c b/hw/intel-hda.c
index 9af27cd..d5ad77c 100644
--- a/hw/intel-hda.c
+++ b/hw/intel-hda.c
@@ -403,7 +403,7 @@ static bool intel_hda_xfer(HDACodecDevice *dev, uint32_t stnr, bool output,
             break;
         }
     }
-    if (st == NULL) {
+    if (s == 4) {
         return false;
     }
     if (st->bpl == NULL) {
-- 
1.7.4.4

